### PR TITLE
Feature/conference api

### PR DESCRIPTION
**Pull request by jonasrob**
##### Retrieve Persisted Conferences - Milestone 1
- Fixed some validation issues with some of the Entities
- Implemented the ConferenceViewBuilder
- Added ConferenceRepository and LocationRepository (might not need)
- Wired up the Conference specific Service methods to the appropriate repository interface instead of the stubFactory

### CODE REVIEW

**Comment by jonasrob** [comment]
To add data needed to make ConferenceView please execute the following queries.

```
insert into conf_admin.location (id, name, short_desc, full_desc, parking_info, create_dttm, last_update_dttm)
select 0, 'Location 0', 'Location: 0', 'Full Description Location: 0', 'The parking information for Location: 0 is the following....', current_date, current_date
```

```
insert into conf_admin.conference(id, name, short_desc, full_desc, start_date, end_date, location_id, 
create_dttm, last_update_dttm)
select 0, 'Conference 0', 'Conf 0 short desc', 'Conference: 0 The complete full description in all of its glory', current_date, current_date, 0, current_date, current_date
```

**Comment by jonasrob** [comment]
#### Update
- Updated the duration in conference_session to integer, and the type in ConferenceSession to Integer.

File: src/main/resources/application.properties [hunk 1]
```diff
@@ -1,5 +1,8 @@
 spring.jpa.hibernate.ddl-auto=validate
-spring.datasource.url=${jdbc.url}
```
Comment (chenry) [inline src/main/resources/application.properties:2]: please bring back the jdbc.url property.  This is intentional.

**Comment by chenry** [comment]
After pulling down the code and running the following:

`mvn clean install -PlocalDB && java -jar target/gs-rest-service-0.1.0.jar`

I still get an exception where I cannot start the application:

`
Caused by: org.hibernate.HibernateException: Wrong column type in conf_admin.conference_session for column room_id. Found: int8, expected: int4
    at org.hibernate.mapping.Table.validateColumns(Table.java:372)
    at org.hibernate.cfg.Configuration.validateSchema(Configuration.java:1336)
    at org.hibernate.tool.hbm2ddl.SchemaValidator.validate(SchemaValidator.java:155)
    at org.hibernate.internal.SessionFactoryImpl.<init>(SessionFactoryImpl.java:523)
    at org.hibernate.cfg.Configuration.buildSessionFactory(Configuration.java:1857)
    at org.hibernate.jpa.boot.internal.EntityManagerFactoryBuilderImpl$4.perform(EntityManagerFactoryBuilderImpl.java:850)
    at org.hibernate.jpa.boot.internal.EntityManagerFactoryBuilderImpl$4.perform(EntityManagerFactoryBuilderImpl.java:843)
    at org.hibernate.boot.registry.classloading.internal.ClassLoaderServiceImpl.withTccl(ClassLoaderServiceImpl.java:399)
    at org.hibernate.jpa.boot.internal.EntityManagerFactoryBuilderImpl.build(EntityManagerFactoryBuilderImpl.java:842)
    at org.hibernate.jpa.HibernatePersistenceProvider.createContainerEntityManagerFactory(HibernatePersistenceProvider.java:150)
    at org.springframework.orm.jpa.LocalContainerEntityManagerFactoryBean.createNativeEntityManagerFactory(LocalContainerEntityManagerFactoryBean.java:336)
    at org.springframework.orm.jpa.AbstractEntityManagerFactoryBean.afterPropertiesSet(AbstractEntityManagerFactoryBean.java:318)
    at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.invokeInitMethods(AbstractAutowireCapableBeanFactory.java:1612)
    at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1549)
    ... 21 more
`

**Comment by chenry** [comment]
done with the review.

**Pull request merged**